src/mpp/Dec.hpp

```diff
@@ -1172,6 +1172,21 @@ bool jump_read_key(BUF& buf, T... t)
 	return jump_find_key<PAIRS, FAMILY, PATH>(val, IS, buf, t...);
 }
 
+template<compact::Family FAMILY, size_t SUBRULE,
+	 class DST, class BUF, class = void>
+struct is_object_readable_by_value : std::false_type {
+};
+
+template<compact::Family FAMILY, size_t SUBRULE,
+	 class DST, class BUF>
+struct is_object_readable_by_value<FAMILY, SUBRULE, DST, BUF,
+				   typename std::enable_if_t<rule_by_family_t<FAMILY>::is_readable_by_value && std::is_assignable_v<DST, decltype(read_value<FAMILY, SUBRULE>(std::declval<BUF>()))>, bool>::type> : std::true_type {
```

Comment:
It seems to me it never works: you already use `std::enable_if_t` and trying to get its `type` field.

Reply:
Fixed. I also forgot to add the lvalue reference to `BUF` in `decltype((read_value<FAMILY, SUBRULE>(std::declval<BUF &>())))`, so it always failed to substitute. I actually checked that it works for optional, I am sorry it slipped through.